docs/guides/mustache.md

```diff
@@ -0,0 +1,112 @@
+
+```graphql
+type Query {
+  paginatedPosts(page: Int!): [Post] @http(path: "/posts?page={{args.page}}")
+}
+```
+
+When executing `paginatedPosts(page: 2)`it constructs a dynamic path: `/posts?page=2`. Here, the Mustache template `/posts?page={{args.page}}` inserts the provided page number directly into the path(in our case 2), guiding the API to retrieve posts from the specified page. This flexibility allows for dynamic path creation, tailored to the required page, through the use of Mustache templates.
+
+#### Dynamic input
+
+Suppose you want to create a post with your input
+
+```graphql
+type Mutation {
+  createPost(input: PostInput!): Post
+    @http(
+      method: "POST"
+      path: "/posts"
+      body: "{{headers.input}}"
+      headers: [{key: "Content-type", value: "application/json; charset=UTF-8"}]
+    )
+}
+
+input PostInput {
+  title: String!
+  body: String!
+  userId: Int!
+}
+```
+
+The createPost mutation needs specific details like title, body, and userId to make a new post.With Mustache's dynamic templating, the `{{headers.input}}` tag inside the body parameter puts the entire input object into the request. This lets us adjust what data gets sent in the POST request.When we use the mutation with the right title, body, and userId, it makes a POST request, creating a new post with the details. Mustache's flexibility helps customize the data sent through the request.To ensure everything's understood, we set the Content-type header to application/json; charset=UTF-8, signaling that the request sends data in JSON format with UTF-8 encoding.
+
+#### Consistency in Configuration
+
+Consider a case where you're utilizing the same base URL across multiple API endpoints.Mustache templates help us to do this in the simplified form below
+
+```graphql
+type Query {
+  users: [User] @http(path: "/users", baseURL: "{{env.API_BASE_URL}}")
+  posts: [Post] @http(path: "/posts", baseURL: "{{env.API_BASE_URL}}")
+}
+```
+
+In this scenario, Mustache templates are employed to integrate environment variables( [env variables in tailcall](environment-variables.md) ), specifically `API_BASE_URL`, into the base URL across multiple API endpoints. By referencing {{env.API_BASE_URL}}, the schema standardizes the base URL for both `users` and `posts` endpoints, ensuring any modifications to the environment variable automatically reflect across all API calls involving these queries. This approach enhances maintainability by allowing global changes to the base URL through the environment variable, promoting consistency in configuration management.
+
+Using Mustache templates in Tailcall lets you create flexible and powerful API configurations. They are like tools that help developers build strong and adaptable systems by allowing dynamic adjustments in how APIs are set up. When used wisely and with smart thinking, Mustache templates make it easier to create strong and adaptable systems for your APIs.
```

Comment:
Before ending, there should be a section of all the places where mustache templates can be used. Refer to the source can add see all the places where it can be used.